src/AutorestSwift/View Models/ServiceClientFileViewModel.swift

```diff
@@ -35,18 +35,46 @@ struct ServiceClientFileViewModel {
     let apiVersionName: String
     let protocols: String
     let paging: Language.PagingNames?
+    let globalParameters: [ParameterViewModel]
+    // A dictionary of all the named operation group. Key is the group name , Value is the operation group view model.
+    let namedOperationGroups: [String: OperationGroupViewModel]
+    // A key,Value pairs of all the named operation group for stencil template engine
+    let namedOperationGroupProperties: [KeyValueViewModel]
 
     init(from model: CodeModel) {
         self.name = "\(model.packageName)Client"
         self.comment = ViewModelComment(from: model.description)
-        var items = [OperationGroupViewModel]()
+        var operationGroups = [OperationGroupViewModel]()
+        var namedOperationGroups = [String: OperationGroupViewModel]()
+        var namedOperationGroupProperties = [KeyValueViewModel]()
         for group in model.operationGroups {
-            items.append(OperationGroupViewModel(from: group, with: model))
+            var viewModel = OperationGroupViewModel(from: group, with: model)
+            var groupName = group.name.trimmingCharacters(in: .whitespacesAndNewlines)
+            if groupName.isEmpty {
+                operationGroups.append(viewModel)
+                // if the operation group name has a collision with a model object, append 'Operation' to the operation group name
+            } else if model.object(for: groupName) != nil {
+                groupName += "Operation"
+                viewModel.name = groupName
+                namedOperationGroups[groupName] = viewModel
+            } else {
+                namedOperationGroups[groupName] = viewModel
+            }
         }
-        self.operationGroups = items
+        self.operationGroups = operationGroups
+        self.namedOperationGroups = namedOperationGroups
         self.apiVersion = model.getApiVersion()
         self.apiVersionName = "v\(apiVersion.replacingOccurrences(of: "-", with: ""))"
         self.paging = model.pagingNames
         self.protocols = paging != nil ? "PipelineClient, PageableClient" : "PipelineClient"
+        var globalParameters = [ParameterViewModel]()
+        for globalParameter in model.globalParameters ?? [] where !globalParameter.name.starts(with: "$") {
+            globalParameters.append(ParameterViewModel(from: globalParameter))
+        }
+        self.globalParameters = globalParameters
+        for key in namedOperationGroups.keys {
+            namedOperationGroupProperties.append(KeyValueViewModel(key: key, value: key.lowercased()))
+        }
+        self.namedOperationGroupProperties = namedOperationGroupProperties
```

Comment:
Is there a reason this appears after global parameters? If so, a comment would be helpful. If not, it might be easier from a maintainability perspective to put this after line 65.

Reply:
Though I'm still not clear why this wouldn't just be located in the OperationGroupViewModel

Reply:
Rename to namedOperationGroupShortcuts